orchestration-shared/src/main/java/uk/gov/di/orchestration/audit/TxmaAuditUser.java

```diff
@@ -1,114 +1,118 @@
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withTransactionId(String transactionId) {
-        this.transactionId = transactionId;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withEmail(String email) {
-        this.email = email;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withPhone(String phone) {
-        this.phone = phone;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withIpAddress(String ipAddress) {
-        this.ipAddress = ipAddress;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withSessionId(String sessionId) {
-        this.sessionId = sessionId;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withPersistentSessionId(String persistentSessionId) {
-        this.persistentSessionId = persistentSessionId;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public TxmaAuditUser withGovukSigninJourneyId(String govukSigninJourneyId) {
-        this.govukSigninJourneyId = govukSigninJourneyId;
-        return this;
+        return new TxmaAuditUser(
+                userId,
+                transactionId,
+                email,
+                phone,
+                ipAddress,
+                sessionId,
+                persistentSessionId,
+                govukSigninJourneyId);
     }
 
     public String getPhone() {
         return phone;
     }
```

Comment:
Why do we need this?

Reply:
Now that it's a record, we don't, good shout. We need it because in `AuditService` we add an extension based on the country code of the phone number. We used to get phone number as a parameter, but now it's on the object.